### DISCUSSION TITLE
Firmware

### DISCUSSION THREAD

**Discussion by valki2** [beeyev/Mikrotik-RouterOS-automatic-backup-and-update#7] [Ideas]
How about adding the option to run a firmware upgrade during the package upgrade?

**Comment by beeyev** [comment]
Actually it should already work, because this feature was implemented from the very beginning.

https://github.com/beeyev/Mikrotik-RouterOS-automatic-backup-and-update/blob/65e4ac4f5827738c988b04b853787d212e87c9a1/BackupAndUpdate.rsc#L318

**Reply by valki2** [reply]
Oh great - didnt catch that one 👍

**Comment by mongoled** [comment]
Sorry to bump this old thread, firstly thank you for maintaining this excellent script over the years.

Could you comment on the reasons why "Part2" of the script is not triggered after the OS has successfully updated and the device has rebooted? 

Ive tried on a SXT Lite5, a SXT Lite5ac and a CRS112-8P-4S. All of them updated the OS to v6.49.15, made back scripts, sent backup scripts to my email, but all of them failed to update the devices firmware.

--edit--
add a hAP lite to that list
